### PR TITLE
Core/Pet: Fix pet action bar after reloading UI

### PR DESCRIPTION
**Pull request by gashole**

### CODE REVIEW

File: src/game/PetAI.cpp [hunk 1]
```diff
@@ -86,6 +86,18 @@ void PetAI::UpdateAI(const uint32 diff)
     else
         m_updateAlliesTimer -= diff;
 
+    // Prevent losing the pet action bar after reloading the user interface
```
Comment (dagochen) [inline src/game/PetAI.cpp:89]: @Zaffy 
You can use this OpCodeHandler, which is called on ReloadUi by the client.
void WorldSession::HandleRequestPetInfoOpcode( WorldPacket & /*recv_data */)
{
    /*
        DEBUG_LOG("WORLD: CMSG_REQUEST_PET_INFO");
        recv_data.hexlike();
    */
}
And simply put in _player->PetSpellInitialize();

Comment (gashole) [reply src/game/PetAI.cpp:89]: Nice, I wasn't aware that anything was called on reload.

Comment (dagochen) [reply src/game/PetAI.cpp:89]: I breakpointed every unhandled opcode :P

And you can also fix the issue for mindcontrol as well by doing this:
if(_player->GetPet())
      _player->PetSpellInitialize();
else if(_player->GetCharm())
      _player->CharmSpellInitialize();

Comment (Lumber-Jack-2) [reply src/game/PetAI.cpp:89]: by @killerwife* :P

Comment (dagochen) [reply src/game/PetAI.cpp:89]: yes sorry for not given credit :(

**Pull request merged**